src/Notes/Notes.php

```diff
@@ -272,16 +272,19 @@ public static function possibly_delete_marketing_notes( $old_value, $value ) {
 	 * Delete actioned survey notes.
 	 */
 	public static function possibly_delete_survey_notes() {
-		$data_store = self::load_data_store();
-		$notes      = $data_store->get_notes(
-			array(
-				'type'   => array( Note::E_WC_ADMIN_NOTE_SURVEY ),
-				'status' => array( 'actioned' ),
+		global $wpdb;
+		$note_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT `note_id` FROM {$wpdb->prefix}wc_admin_notes WHERE `type` = %s AND `status` = %s",
```

Comment:
I think this is confirming your suspicions about circumventing the `get_notes()` call and expectations around filters.

I would argue that this is a third party issue and not one we should be manual querying around.